src/plugins/data/server/index_patterns/fetcher/lib/opensearch_api.ts

```diff
@@ -57,10 +57,20 @@ export interface IndexAliasResponse {
  *  @return {Promise<IndexAliasResponse>}
  */
 export async function callIndexAliasApi(
-  callCluster: LegacyAPICaller,
+  callCluster: LegacyAPICaller | OpenSearchClient,
   indices: string[] | string
 ): Promise<IndicesAliasResponse> {
   try {
+    if ('transport' in callCluster) {
```

Comment:
A comment here would help developers know the significance of having `callCluster.transport`

Reply:
Will update, also adding the migration task for support data client with legacy client.

Reply:
we have to pending PR to add support for legacy client. After that PR is in, this will be removed
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2133